generators/client/templates/angular/src/main/webapp/app/admin/audits/audits.component.ts.ejs

```diff
@@ -30,113 +29,109 @@ import { AuditsService } from './audits.service';
-        today.setDate(today.getDate() + 1);
-        const date = new Date(today.getFullYear(), today.getMonth(), today.getDate());
-        this.toDate = this.datePipe.transform(date, dateFormat);
+        const date = new Date();
+        date.setDate(date.getDate() + 1);
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.toDate = this.datePipe.transform(date, this.dateFormat)!;
     }
 
-    loadAll() {
-        this.auditsService.query({
-            page: this.page - 1,
-            size: this.itemsPerPage,
-            sort: this.sort(),
-            fromDate: this.fromDate,
-            toDate: this.toDate}).subscribe(
-            (res: HttpResponse<Audit[]>) => this.onSuccess(res.body, res.headers),
-            (res: HttpResponse<any>) => this.onError(res.body)
-        );
+    private loadData(): void {
+        this.auditsService
+            .query({
+                page: this.page - 1,
+                size: this.itemsPerPage,
+                sort: this.sort(),
+                fromDate: this.fromDate,
+                toDate: this.toDate
+            })
+            .subscribe((res: HttpResponse<Audit[]>) => this.onSuccess(res.body, res.headers), () => (this.initialized = true));
     }
 
-    sort() {
-        const result = [this.predicate + ',' + (this.reverse ? 'asc' : 'desc')];
+    private sort(): string[] {
+        const result = [this.predicate + ',' + (this.ascending ? 'asc' : 'desc')];
         if (this.predicate !== 'id') {
             result.push('id');
         }
         return result;
     }
 
-    loadPage(page: number) {
+    loadPage(page: number): void {
         if (page !== this.previousPage) {
             this.previousPage = page;
             this.transition();
         }
     }
 
-    transition() {
+    transition(): void {
+        this.canLoad = this.fromDate !== '' && this.toDate !== '';
+        if (!this.canLoad) {
+            return;
+        }
         this.router.navigate(['/admin/audits'], {
             queryParams: {
                 page: this.page,
-                sort: this.predicate + ',' + (this.reverse ? 'asc' : 'desc')
+                sort: this.predicate + ',' + (this.ascending ? 'asc' : 'desc')
             }
         });
-        this.loadAll();
-    }
-
-    private onSuccess(data, headers) {
-        this.links = this.parseLinks.parse(headers.get('link'));
-        this.totalItems = headers.get('X-Total-Count');
-        this.audits = data;
+        this.loadData();
     }
 
-    private onError(error) {
-        this.alertService.error(error.error, error.message, null);
+    private onSuccess(audits: Audit[] | null, headers: HttpHeaders): void {
+        this.totalItems = Number(headers.get('X-Total-Count'));
+        if (audits === null) {
+            this.audits = [];
+        } else {
+            this.audits = audits;
+        }
+        this.dataHaveBeenLoaded = true;
+        this.initialized = true;
```

Comment:
What's the difference between dataHaveBeenLoaded and initialized ?
Btw it's dataHasBeenLoaded I think 😄

Reply:
`initialized` is valued to `true` also then if error occurs on data loading. So if on first page load error occurs then anyway date filter is shown and user can try to change dates and retry data loading in this way.

Reply:
I moved `initialized = true` to `ngOnInit` now.  
It aligns initial behavior with further behavior: running new query before previous query is not completed is now allowed also while first query is running.  
And code looks better after this change.  
I force pushed this change.